### PR TITLE
resolves 168

### PR DESCRIPTION
**Pull request by Sn1p3rr3c0n**
Creating the Pull request in Preparation. **Not ready to merge!**

as of now it adds the items to the total Counts. Allowing the use of produce till x Amount even if the output is blocked.

### CODE REVIEW

**Comment by Sn1p3rr3c0n** [comment]
@zymex22 most things should be in now. Can you do some tests and provided feedback if additional enhancements are needed?

**Comment by zymex22** [comment]
- count towards total working and NICE addition! :D 
- i like that xx/100 buffer and working
- label when 100/100 working.


anyways we might add a "xx wating to be placed" on second line when there is more than 1 items in queqe,
it took me forever to till up to 100 with stone cutter so that would have taking me some time to figure something is wrong without looking at info box.

like this maybe:
image

File: Source/ProjectRimFactory/Common/HarmonyPatches/UpdateResourceCountsPatch.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using System.Threading.Tasks;
+
+namespace ProjectRimFactory.Common.HarmonyPatches
+{
+
+
+    [StaticConstructorOnStartup]
+    public static class AssemblerQueueManager
+    {
+        public static List<IAssemblerQueue> AssemblerQueue = new List<IAssemblerQueue>();
```
Comment (lilwhitemouse) [inline Source/ProjectRimFactory/Common/HarmonyPatches/UpdateResourceCountsPatch.cs:18]: This won't reset on game restart or anything like that.

Prefer to add the AssemblerQueue to the PRFGameComponent as a local variable - that way, it's tied to the specific game.

File: Source/ProjectRimFactory/SAL3/Things/Assemblers/Building_ProgrammableAssembler.cs [hunk 2]
```diff
@@ -539,6 +543,10 @@ public List<Thing> GetThingQueue()
         // thingQueue is List to save properly
         //   List of produced things, waiting to be placed:
         protected List<Thing> thingQueue = new List<Thing>();
+        //max number of items that can be stored in thingQueue before production is halted
+        protected const int max_thingQueue_Count = 100;
+
+        protected bool allowProduction_thingQueue => thingQueue.Count < max_thingQueue_Count;
```
Comment (lilwhitemouse) [inline Source/ProjectRimFactory/SAL3/Things/Assemblers/Building_ProgrammableAssembler.cs:561]: C# Convention would be AllowProduction_thingQueue => etc (capital letter; reserve lowercase for actual member fields)

**Pull request merged**